3G Metabarcoding
===========================

Lecture
^^^^^^^

.. slide:: https://docs.google.com/presentation/d/1lYXYCYSYASscGGCTzT8QBm6tZHFDwb_ELCk8-sfoaRU

Introduction
^^^^^^^^^^^^
Here, we will analyze a non-routine diagnostic rice sample using 2G metabarcoding. The sample tested positive for Pantoea ananatis using a conventional PCR assay; however, qPCR data for these samples was not available.  A highly degenerate primer pair was used to amplify an ~290 bp fragment of DNA gyrase ß subunit gene (gyrB), a common phylogenetic marker present in single copy in most bacteria. Amplicons were sequenced by 2x250 Illumina MiSeq sequencing.

Identifying bacteria from amplicon sequencing data can be done with two main techniques:

* Cluster sequences into Operational Taxonomic Units (OTUs)
* Reconstruct exact sequences into Amplicon Sequence Variants (ASVs)

In this tutorial, we will focus on ASV analysis using the popular R package DADA2 https://benjjneb.github.io/dada2/.

Import Data
^^^^^^^^^^^

Lets import data from a shared history. These are raw reads, exactly how you would receive them from a sequencing company or off your own Illumina sequencer. Files that have been split (or “demultiplexed”) by sample and the barcodes/adapters have already been removed by the sequencer. There is also a curated database of ~40,000 gyrB reference sequences, which allows for the taxonomic annotation of a gyrB amplicon dataset. This database was generated by Barret et al. (2015).

.. admonition:: Hands-On: Import Metagenomic Reads from Shared History

    1. At the top of the screen click on ``Shared Data`` and select ``Pages``

    2. In the search field, search for ``NPDN`` and click on the page ``NPDN HTS 2023``

    3. Find the history for ``NPDN 2023 Data 2G Metabarcode`` Select the green plus sign to import into your Galaxy environment.